syntax = "proto3";

//import "grpcpp/cpp_options.proto";   // 必须导入，否则报错
//option cc_namespace = "my::rpc";     // ② 把 C++ 命名空间强制改成 my::rpc

package routeguide;

message Request {
    int32 id = 1;
    bytes name = 2;
}

message Response {
    string message = 1;
}

service RouteGuide {
    // 一元RPC
    rpc sayHello (Request) returns(Response) {};

    // 服务器流RPC
    rpc ListFeatures(Request) returns (stream Response) {}

    // 客户端流PRC
    rpc RecordRoute(stream Request) returns (Response) {}

    // 双向流RPC
    rpc RouteChat(stream Request) returns (stream Response) {}
}
